Tidbit/StorageAdapter/Storage/Oracle.php

```diff
@@ -79,4 +81,66 @@ protected function prepareQuery($tableName, array $installData)
 
         return $sql . ' SELECT * FROM dual';
     }
+
+    /**
+     * Patch inserted value if it look like sequence
+     *
+     * @param array $installData
+     */
+    protected function patchSequenceValues(array &$installData)
+    {
+        if (!$sequence = $this->getSequenceFromValues($installData[0])) {
+            return;
+        }
+
+        $currentValue = $this->getCurrentSequenceValue($sequence['name']);
+        $installDataCount = count($installData);
+        for ($i=0; $i <$installDataCount; $i++) {
+            $installData[$i][$sequence['field']] = $currentValue++;
+        }
+
+        $this->setNewSequenceValue($sequence['name'], $installDataCount);
+    }
+
+    /**
+     * Check array of values on containing sequence value
+     *
+     * @param array $values
+     * @return array
+     */
+    protected function getSequenceFromValues(array $values)
+    {
+        foreach ($values as $k => $v) {
+            if (substr($v, -12) == '_SEQ.NEXTVAL') {
+                return ['field' => $k, 'name' => substr($v, 0, -8)];
+            }
+        }
+        return [];
+    }
+
+    /**
+     * rtfn
+     *
+     * @param string $sequenceName
+     * @return int
+     */
+    protected function getCurrentSequenceValue($sequenceName)
+    {
+        $sql = sprintf("SELECT %s.NEXTVAL as current_val FROM dual", $sequenceName);
+        $result = $this->storageResource->query($sql);
+        $row = $this->storageResource->fetchByAssoc($result);
+        return intval($row['current_val']);
+    }
+
+    /**
+     * rtfn
+     *
+     * @param string $sequenceName
+     * @param int $incrementOn
+     */
+    protected function setNewSequenceValue($sequenceName, $incrementOn)
+    {
+        $sql = sprintf("ALTER SEQUENCE %s INCREMENT BY %d", $sequenceName, $incrementOn);
```

Comment:
we should call SEQ_NAME.NEXVAL and return back INCREMENT to 1 after this action, see example on jira